test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/DefaultModelMetadataProviderTest.cs

```diff
@@ -86,6 +86,7 @@ public void GetMetadataForProperties_Cached()
             // Assert
             for (var i = 0; i < metadata1.Length; i++)
             {
+                Assert.Same(metadata1[0], metadata2[1]);
```

Comment:
Updated this code:

```
// Act
            var properties1 = provider.GetMetadataForProperties(typeof(ModelType)).Cast<DefaultModelMetadata>().ToArray();
            var properties2 = provider.GetMetadataForProperties(typeof(ModelType)).Cast<DefaultModelMetadata>().ToArray();

            // Assert
            Assert.Equal(properties1.Length, properties2.Length);
            for (var i = 0; i < properties1.Length; i++)
            {
                Assert.Same(properties1[i], properties2[i]);
                Assert.Same(properties1[i].Attributes, properties2[i].Attributes);
                Assert.Same(properties1[i].BindingMetadata, properties2[i].BindingMetadata);
                Assert.Same(properties1[i].DisplayMetadata, properties2[i].DisplayMetadata);
                Assert.Same(properties1[i].ValidationMetadata, properties2[i].ValidationMetadata);
            }
```

All the assertions for the individual properties (BindingMetadata, etc) are still important - they verify that we cache these.